simpeg/utils/mesh_utils.py

```diff
@@ -94,3 +98,187 @@ def surface2inds(vrtx, trgl, mesh, boundaries=True, internal=True):
 
     # Return the indexes inside
     return insideGrid
+
+
+def _closest_grid_indices(grid, pts, dim=2):
+    """Return indices of closest gridded points for a set of input points.
+
+    Parameters
+    ----------
+    grid : (n, dim) numpy.ndarray
+        A gridded set of points.
+    pts : (m, dim) numpy.ndarray
+        Points being projected to gridded locations.
+    dim : int
+        The `dim` input is deprecated and will be removed in SimPEG v0.24.0.
+
+    Returns
+    -------
+    (n,) numpy.ndarray
+        Indices of the closest gridded points for all *pts* supplied.
+    """
```

Comment:
I think we can remove the `dim` argument, right?
```suggestion
def _closest_grid_indices(grid, pts):
    """Return indices of closest gridded points for a set of input points.

    Parameters
    ----------
    grid : (n, dim) numpy.ndarray
        A gridded set of points.
    pts : (m, dim) numpy.ndarray
        Points being projected to gridded locations.

    Returns
    -------
    (n,) numpy.ndarray
        Indices of the closest gridded points for all *pts* supplied.
    """
```